test.js

```diff
@@ -25,10 +25,86 @@ test('number option', validate, {
 	number: ['score']
 });
 
-test('array option', validate, {
+test('default array option', validate, {
 	arr: 'array'
 }, {
-	array: ['arr']
+	array: [{key: 'arr', string: true}]
+});
+
+test('string array option', validate, {
+	arr: ['string']
```

Comment:
Could you change the way of specifying arrays to `string-array`, `boolean-array` and `number-array`, as you've described in https://github.com/vadimdemedes/minimist-options/issues/18? `['string']` is odd to me, because it feels like TypeScript when it's not. I think this should simplify the source code too.

Reply:
Sure. I agree that is a little weird, but I couldn't resist its beauty.

But you're right. It would make the implementation easier and I guess there's not more magic string-shenanigans going on here than there already are in the other types.

Reply:
The only thing that's weird now is having both `array` and `string-array`. I would vote for deprecating `array` in favor of `string-array`.

Thoughts?

Reply:
But then again, `yargs(-parser)` supports both. Maybe leave it as is for now?

Reply:
Left as is for now